lib/dry/types/struct.rb

```diff
@@ -87,10 +89,9 @@ def [](name)
       end
 
       def to_hash
-        self.class.schema.keys.each_with_object({}) { |key, result|
-          value = self[key]
-          result[key] = value.respond_to?(:to_hash) ? value.to_hash : value
-        }
+        self.class.schema.keys.each_with_object({}) do |key, result|
+          result[key] = Hashify[self[key]]
```

Comment:
it's an overkill to instantiate Hashify object here, could we simply provide this functionality in this method instead?